src/python/pants/backend/docker/target_types.py

```diff
@@ -30,8 +33,23 @@ class DockerBuildArgsField(StringSequenceField):
 
 class DockerImageSourceField(SingleSourceField):
     default = "Dockerfile"
+    default_glob_match_error_behavior = GlobMatchErrorBehavior.ignore
+    expected_num_files = range(0, 2)
+    required = False
+    help = (
+        "The Dockerfile to use when building the Docker image.\n\n"
+        "Use the `instructions` field instead if you prefer not having the Dockerfile in your "
+        "project source tree."
+    )
+
+
+class DockerImageInstructionsField(StringSequenceField):
+    alias = "instructions"
     required = False
-    help = "The Dockerfile to use when building the Docker image."
+    help = (
+        "The `Dockerfile` content, typically one instruction per list item.\n\n"
```

Comment:
For symmetry with the phrasing above, instead of the "mutually exclusive" part, maybe say ``"Use the `source` field instead if you prefer having the
Dockerfile in your project source tree."`` The mutual exclusion is implied by the "instead".